### PR TITLE
.*: fix bug introduced by https://github.com/pingcap/tiflash/pull/7175

### LINKED ISSUE #7229: Unstable unit test: FineGrainedShuffleTestRunner.simpleReceiver
https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tiflash-ghpr-unit-tests/runs/9253/nodes/75/steps/80/log/?start=0
```
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.662 +08:00] [DEBUG] [Event.cpp:143] ["one task finished, 0 tasks remaining"] [source="executor_test 0"] [thread_id=51]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.663 +08:00] [DEBUG] [Event.cpp:151] ["switch status: SCHEDULED --> FINISHED"] [source="executor_test 0"] [thread_id=51]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.663 +08:00] [DEBUG] [Event.cpp:151] ["switch status: INIT --> SCHEDULED"] [source="executor_test 0"] [thread_id=1]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.663 +08:00] [DEBUG] [Event.cpp:133] ["1 tasks scheduled by event"] [source="executor_test 0"] [thread_id=1]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.663 +08:00] [TRACE] [Task.h:113] ["switch status: INIT --> RUNNING"] [source="executor_test 0"] [thread_id=1]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.664 +08:00] [TRACE] [Task.h:113] ["switch status: RUNNING --> FINISHED"] [source="executor_test 0"] [thread_id=43]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.664 +08:00] [DEBUG] [Event.cpp:143] ["one task finished, 0 tasks remaining"] [source="executor_test 0"] [thread_id=43]
[2023-04-06T07:58:17.228Z] [2023/04/06 15:58:14.664 +08:00] [DEBUG] [Event.cpp:151] ["switch status: SCHEDULED --> FINISHED"] [source="executor_test 0"] [thread_id=43]
[2023-04-06T07:58:17.228Z] libc++abi: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
```

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #7229

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Flash/Executor/PipelineExecutor.cpp [hunk 1]
```diff
@@ -35,13 +35,13 @@ void PipelineExecutor::scheduleEvents()
 {
     assert(root_pipeline);
     auto events = root_pipeline->toEvents(status, context, context.getMaxStreams());
-    Events without_input_events;
+    Events sources;
     for (const auto & event : events)
     {
-        if (event->withoutInput())
-            without_input_events.push_back(event);
+        if (event->prepreForSource())
```
Comment (windtalker) [inline dbms/src/Flash/Executor/PipelineExecutor.cpp:41]: You mean `prepareForSource`?

File: dbms/src/Flash/Pipeline/Schedule/Events/tests/gtest_event.cpp [hunk 2]
```diff
@@ -499,7 +500,8 @@ try
     {
         PipelineExecutorStatus exec_status;
         auto event = std::make_shared<ManyTasksEvent>(exec_status, i);
-        event->schedule();
+        if (event->prepreForSource())
+            event->schedule();
```
Comment (windtalker) [inline dbms/src/Flash/Pipeline/Schedule/Events/tests/gtest_event.cpp:504]: Before this pr, all the event will be scheduled without check the dependency?

File: dbms/src/Flash/Executor/PipelineExecutor.cpp (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Flash/Executor/PipelineExecutor.cpp:41]: yes...typo fixed

File: dbms/src/Flash/Pipeline/Schedule/Events/tests/gtest_event.cpp (hunk 2 above)
Comment (SeaRise) [reply dbms/src/Flash/Pipeline/Schedule/Events/tests/gtest_event.cpp:504]: Before this pr, only event can be schedule immediately `event.withoutInput` == true.
But it doesn't matter if `event.withoutInput` doesn't call.
In this pr, `event.prepareForSource` must be called before `event.schedule`

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**